### DISCUSSION TITLE
Why do non-string scalars in Server-invoked DDL-Explicit Default Expressions require `text` and a quote?

### DISCUSSION THREAD

**Discussion by whatnot-jeffavery** [sqlalchemy/sqlalchemy#8334] [Usage Questions]
I am adding a column with a scalar server_default  (`0`). From reading the [docs](https://docs.sqlalchemy.org/en/14/core/defaults.html#server-invoked-ddl-explicit-default-expressions), it seems that using `text` is advised.

`t = Table('test', metadata_obj,
    Column('abc', String(20), server_default='abc'),
    Column('created_at', DateTime, server_default=func.sysdate()),
    Column('index_value', Integer, server_default=text("0"))
)`

> It is advisable to use the [text()](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text) construct for any literal SQL values as opposed to passing the raw value, as SQLAlchemy does not typically perform any quoting or escaping on these values.

Why would SQLAlchemy need to perform quoting or escaping of these values, especially if it's intended to be a non-string scalar? Is this a design decision or simply not a common enough use-case to support non-string scalars for server_default values?

**Comment by zzzeek** [comment]
SQLAlchemy historically has never implemented client-side rendering of SQL literal values, which are normally passed as bound parameters in statements where they are handled by the DBAPI in use, so by providing text() this ensures that the person creating the schema can ensure correct SQL representation as per the backend in use.

These days SQLAlchemy does in fact have client side SQL renderers for a lot of literal types but for server DDL we still standardize on a SQL expression, not a Python literal value, being passed.
